### PR TITLE
refactor(xbps): clean up error handling logic

### PR DESCRIPTION
**Pull request by rami3l**
This is a follow-up PR simplifying the error handling logic for `xbps` proposed in #330.

cc @tranzystorek-io

### CODE REVIEW

File: src/pm/xbps.rs [hunk 1]
```diff
@@ -84,23 +84,21 @@ impl Pm for Xbps {
             .await?;
 
         let mut stdout = std::io::stdout();
-        let mut res = Ok(());
-        for line in lines {
-            match line {
-                Ok(line) => stdout.write_all(&line)?,
+        lines.into_iter().try_fold(Ok(()), |acc, line| {
```
Comment (tranzystorek-io) [inline src/pm/xbps.rs:87]: Looks like a job for `try_for_each` instead 🙂

File: src/pm/xbps.rs [hunk 2]
```diff
@@ -140,23 +138,21 @@ impl Pm for Xbps {
...
```
Comment (tranzystorek-io) [inline src/pm/xbps.rs:141]: Same as above

File: src/pm/xbps.rs [hunk 3]
```diff
@@ -84,23 +84,21 @@ impl Pm for Xbps {
...
+            std::io::Result::Ok(match line {
```
Comment (tranzystorek-io) [inline src/pm/xbps.rs:88]: I don't think you need to specify the exact result type for `Ok`

File: src/pm/xbps.rs (hunk 1 above)
Comment (tranzystorek-io) [reply src/pm/xbps.rs:87]: Nevermind I noticed you're folding the result 😅

File: src/pm/xbps.rs (hunk 3 above)
Comment (rami3l) [reply src/pm/xbps.rs:88]: @tranzystorek-io 
I agree with you, yet it's the Rust compiler that enforces it, not me :)

**Pull request merged**